core/state.py

```diff
@@ -1366,6 +1366,10 @@ def GetArg0(self):
     def GetArgNum(self, arg_num):
         # type: (int) -> value_t
         if arg_num == 0:
+            # $0 may be overriden, eg. by Str => replace()
+            if "0" in self.var_stack[-1] and (
```

Comment:
I think you could have a variable `regex_match = self.var_stack[-1].get('0')` and then this subexpression would only appear once,

I think the code would be slightly shorter